tests/test_peer_did.py

```diff
@@ -1,13 +1,108 @@
+
+
+def test_create_numalgo_2_positive():
+    encryption_keys = [PublicKeyAgreement(encoded_value="6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+                                          type=KeyTypeAgreement.X25519)]
+    signing_keys = [
+        PublicKeyAuthentication(
+            encoded_value="zXwpBnMdCm1cLmKuzgESn29nqnonp1ioqrQMRHNsmjMypp",
+            type=KeyTypeAuthentication.ED25519),
+        PublicKeyAuthentication(
+            encoded_value="zx8xB2pv7cw8q1PdDacSrdWE3dtB9f7Nxk886mdzNFoPtY",
+            type=KeyTypeAuthentication.ED25519)]
+    service = '''{
+        "type": "didcommmessaging",
+        "serviceEndpoint": "https://example.com/endpoint",
+        "routingKeys": ["did:example:somemediator#somekey"]
+        }
+        '''
+
+    peer_did_algo_2 = create_peer_did_numalgo_2(encryption_keys=encryption_keys, signing_keys=signing_keys,
+                                                service=service)
+
+
+def test_create_numalgo_2_wrong_encryption_key():
+    encryption_keys = [PublicKeyAgreement(encoded_value="....",
+                                          type=KeyTypeAgreement.X25519)]
+    signing_keys = [
+        PublicKeyAuthentication(
+            encoded_value="zXwpBnMdCm1cLmKuzgESn29nqnonp1ioqrQMRHNsmjMypp",
+            type=KeyTypeAuthentication.ED25519),
+        PublicKeyAuthentication(
+            encoded_value="zx8xB2pv7cw8q1PdDacSrdWE3dtB9f7Nxk886mdzNFoPtY",
+            type=KeyTypeAuthentication.ED25519)]
+    service = '''{
+        "type": "didcommmessaging",
+        "serviceEndpoint": "https://example.com/endpoint",
+        "routingKeys": ["did:example:somemediator#somekey"]
+        }
+        '''
+
+    with pytest.raises(ValueError):
+        peer_did_algo_2 = create_peer_did_numalgo_2(encryption_keys=encryption_keys, signing_keys=signing_keys,
+                                                    service=service)
+
+
+def test_create_numalgo_2_wrong_signing_key():
+    encryption_keys = [PublicKeyAgreement(encoded_value="6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+                                          type=KeyTypeAgreement.X25519)]
+    signing_keys = [
+        PublicKeyAuthentication(
+            encoded_value=".......",
+            type=KeyTypeAuthentication.ED25519),
+        PublicKeyAuthentication(
+            encoded_value="zx8xB2pv7cw8q1PdDacSrdWE3dtB9f7Nxk886mdzNFoPtY",
+            type=KeyTypeAuthentication.ED25519)]
+    service = '''{
+        "type": "didcommmessaging",
+        "serviceEndpoint": "https://example.com/endpoint",
+        "routingKeys": ["did:example:somemediator#somekey"]
+        }
+        '''
+
+    with pytest.raises(ValueError):
+        peer_did_algo_2 = create_peer_did_numalgo_2(encryption_keys=encryption_keys, signing_keys=signing_keys,
+                                                    service=service)
+
+
+def test_create_numalgo_2_wrong_service():
```

Comment:
We should also add a test that raises an error when PublicKeyAgreement is passed a signing key and/or  PublicKeyAuthentication is passed as an authentication key